src/sentry/api/event_search.py

```diff
@@ -421,6 +430,20 @@ def visit_time_filter(self, node, children):
             search_value = operator + search_value if operator != "=" else search_value
             return self._handle_basic_filter(search_key, "=", SearchValue(search_value))
 
+    def visit_duration_filter(self, node, children):
+        (search_key, _, operator, search_value) = children
+
+        operator = operator[0] if not isinstance(operator, Node) else "="
+        if search_key.name in self.duration_keys:
+            try:
+                search_value = parse_duration(*search_value.match.groups())
+            except InvalidQuery as exc:
+                raise InvalidSearchQuery(six.text_type(exc))
```

Comment:
It would be good to have some tests around these failure scenarios as we shouldn't get 500s from them.